server/src/main/java/org/opensearch/common/settings/WriteableSetting.java

```diff
@@ -138,6 +151,7 @@ private Setting<?> createSetting(
         SettingType type,
         String key,
         Object defaultValue,
+        Object parser,
```

Comment:
You need to use the values from this parser when you are creating the settings in the switch/case on line 158-197.

For example for the `case Integer` you will extract `((IntegerParser) parser).getMin()` and `((IntegerParser) parser).getMax()` and add them as parameters to `Setting.intSetting()`.  And so on for the rest of the cases.